### DISCUSSION TITLE
Upgrade to latest version of a specific major version

### DISCUSSION THREAD

**Discussion by izzap** [chocolatey/choco#2447] [Q&A]
Is it possible to upgrade to the latest released version of a package with a specific major version? Similarly to npm?

`choco upgrade my-awesome-package --version ^1.1.1` 

This should get the latest release version of major 1

I need this feature when package A depends on package B, but B has a newer minor or patch release after release of A

**Comment by TheCakeIsNaOH** [comment]
This is not directly possible yet, but it is something that is planned to be added: https://github.com/chocolatey/choco/issues/800

It is possible via a wrapper script, see this example for python3: https://github.com/chocolatey-community/chocolatey-packages/issues/1602#issuecomment-941673094
